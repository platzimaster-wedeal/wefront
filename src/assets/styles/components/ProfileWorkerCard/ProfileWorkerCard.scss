@import '../../partials/colors.scss';
@import '../../partials/sizes.scss';
@import '../../Breakpoints.scss';

.profile-worker-card {
  box-sizing: border-box;
  display: flex;
  position: relative;
  flex-flow: row nowrap;
  align-items: center;
  width: 90%;
  padding: $space-size-small;
  margin: $space-size-small 0;
  background-color: #fff;
  border-top: 1px solid $color-black-30;
  border-bottom: 1px solid $color-black-30;
  cursor: pointer;

  &:hover .profile-worker-card__content-action {
    opacity: 1;
  }

  &__picture {
    margin: 0;
    padding: 0;
    margin-right: $space-size-small;
  }

  &__content {
    position: relative;
    width: 100%;
    display: flex;
    flex-flow: column nowrap;
  }
  &__content-header {
    width: 100%;
    display: flex;
    padding: $space-size-extra-small 0;
    border-bottom: 3px solid $color-primary;

    & h3 {
      color: $color-primary;
      @include font-size($font-size-normal);
      margin: 0;
      padding: 0;

      @include media(tablet) {
        @include font-size($font-size-medium);
      }
    }
  }
  &__content-action {
    position: absolute;
    right: 3%;
    top: 30%;    
    opacity: 0;
    transition: .6s ease-in-out;

    @include media(tablet) {
      right: 3%;
      top: 9%;
    } 
  }

  &__description {
    padding: 3px 0;
  }

  &__stats {
    display: flex;
    flex-flow: row nowrap;
  }

}
