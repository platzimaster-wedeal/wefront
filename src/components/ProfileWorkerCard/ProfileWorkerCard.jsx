import React from "react";
import { MdRemoveRedEye, MdPeople } from "react-icons/md";

// Imgs
import PhotoUser from "../../assets/static/img/users/user.jpg";

// Stlyes
import "../../assets/styles/components/ProfileWorkerCard/ProfileWorkerCard.scss";

// Components
import GraphicData from "../GraphicData/GraphicData";
import ProfilePicture from "../ProfilePictures/ProfilePicture";
import ButtonAction from "../Buttons/ButtonAction";

const ProfileWorkerCard = ({
 picture,
 name = "Tupac Shakur",
 profession = "Artist",
 description,
 follows = 24,
 onClick = () => {}
}) => {
 return (
  <article className="profile-worker-card">
   <figure className="profile-worker-card__picture">
    <ProfilePicture source={picture || PhotoUser} alt={name} />
   </figure>

   <div className="profile-worker-card__content">
    <div className="profile-worker-card__content-header">
     <h3>{name}</h3>
     <small>{profession}</small>
    </div>

    <ButtonAction
     icon={MdRemoveRedEye}
     className="profile-worker-card__content-action"
     onClick={onClick}
    />

    <small className="profile-worker-card__description">
     {description || "Hi there, I'm Tupac :D"}
    </small>

    <div className="profile-worker-card__stats">
     <GraphicData active value={follows}>
      <MdPeople size="24" />
     </GraphicData>
    </div>
   </div>
  </article>
 );
};

export default ProfileWorkerCard;
