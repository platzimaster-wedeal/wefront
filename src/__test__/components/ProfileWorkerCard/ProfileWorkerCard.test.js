import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import ProfileWorkerCard from "../../../components/ProfileWorkerCard/ProfileWorkerCard";

describe("Tests for ProfileWorkerCard components", () => {
 const ProfileWorkerCardMounted = mount(<ProfileWorkerCard />);
 const ProfileWorkerCardCreated = create(<ProfileWorkerCard />);

 describe("Visual tests for the SelectCategoires component", () => {
  test("should the component render one component", () => {
   expect(ProfileWorkerCardMounted.length).toEqual(1);
  });
  test("should the component have the same JSON tree", () => {
   expect(ProfileWorkerCardCreated.toJSON).toMatchSnapshot();
  });

  test("should the name of the component be a string", () => {
   expect(
    typeof ProfileWorkerCardMounted.find(
     ".profile-worker-card__content-header h3"
    ).text()
   ).toEqual("string");
  });
  test("should the profession of the component be a string", () => {
   expect(
    typeof ProfileWorkerCardMounted.find(
     ".profile-worker-card__content-header small"
    ).text()
   ).toEqual("string");
  });

  test("should the description of the component be a string", () => {
   expect(
    typeof ProfileWorkerCardMounted.find(
     ".profile-worker-card__description"
    ).text()
   ).toEqual("string");
  });
 });
});
