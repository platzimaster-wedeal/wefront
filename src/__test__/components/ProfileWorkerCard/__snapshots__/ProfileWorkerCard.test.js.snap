// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for ProfileWorkerCard components Visual tests for the SelectCategoires component should the component have the same JSON tree 1`] = `[Function]`;
